presto-hive/src/main/java/com/facebook/presto/hive/ParquetHiveRecordCursor.java

```diff
@@ -709,7 +709,23 @@ public ParquetListConverter(String columnName, GroupType listType)
                     columnName,
                     listType.getFieldCount());
 
-            elementConverter = new ParquetListEntryConverter(columnName, listType.getType(0).asGroupType());
+            // Hive serializes parquet lists as a repeated group with one
+            // optional array element, to support nulls, like so:
+            //
+            // optional group listField (LIST) {
+            //   repeated group bag {
+            //     optional int array_element
+            //   }
+            // }
+            //
+            // Other parquet writers omit this level of indirection.
+            parquet.schema.Type elementType = listType.getType(0);
+            if (elementType.getName().equals("bag") && !elementType.isPrimitive()) {
```

Comment:
I think we should remove the check that the name is "bag". The spec you linked to says: "these names may not be used in existing data and should not be enforced as errors when reading"